src/CPUSolver.cpp

```diff
@@ -652,16 +649,9 @@ FP_PRECISION CPUSolver::computeFSRSources() {
     else
       fission_source = 0.0;
 
-    /* Compute total scattering source for group G */
     for (int G=0; G < _num_groups; G++) {
-      scatter_source = 0;
-
-      for (int g=0; g < _num_groups; g++)
-        _scatter_sources(tid,g) = material->getSigmaSByGroupInline(g,G)
-                      * _scalar_flux(r,g);
-
-      scatter_source=pairwise_sum<FP_PRECISION>(&_scatter_sources(tid,0),
-                                                _num_groups);
+      /* Compute total scattering source for group G */
+      scatter_source = material->getScatterSource(G,&_scalar_flux(r,0));
```

Comment:
I like the thought, but the implementation is a little tricky, as we precompute the total fission source and then multiply by chi.  So maybe the routine should be to compute the total fission source (not group specific) and then just multiply by chi in the solver function?